client/src/components/Tea/TeaCollectionTable/TeaCollectionTableContainer.jsx

```diff
@@ -0,0 +1,218 @@
+      this.state.formControls.filterCriteria !== ""
+    ) {
+      newFilterOrder = list.filter(item => {
+        if (Number(this.state.formControls.filterCriteria)) {
+          return (
+            this.props.teas.allTeas[item][
+              this.state.formControls.filterCategory
+            ] == this.state.formControls.filterCriteria
+          );
+        } else {
+          return (
+            this.props.teas.allTeas[item][
+              this.state.formControls.filterCategory
+            ]
+              .toLowerCase()
+              .search(this.state.formControls.filterCriteria.toLowerCase()) !==
+            -1
+          );
+        }
+      });
+      const revSortOrder = [...newFilterOrder].reverse();
+
+      currentFilterState = {
+        asc: newFilterOrder,
+        desc: revSortOrder
+      };
+
+      this.setState({
+        sortedIDs: newFilterOrder,
+        memoizedIDs: currentFilterState,
+        filtered: true
+      });
+    }
+  };
+
+  handleClearFilterClick = () => {
+    this.setState({
+      sortedIDs: this.props.teas.teaIDs,
+      memoizedIDs: this.props.teas.teaIDs,
+      formControls: {
+        filterCategory: "",
+        filterCriteria: ""
+      },
+      filtered: false,
+      dataList: []
+    });
+  };
+
+  componentDidMount() {
+    this.props.getTeaList(this.props.userID);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.teas.teaIDs &&
+      this.props.teas.teaIDs !== prevProps.teas.teaIDs
+    ) {
+      this.setState({
+        sortedIDs: this.props.teas.teaIDs,
+        memoizedIDs: {},
+        sortColumn: "",
+        sortOrder: "",
+        formControls: {
+          filterCategory: "",
+          filterCriteria: ""
+        },
+        filtered: false
+      });
+    }
+  }
+
+  render() {
+    return (
+      this.props.teas.allTeas && (
+        <TeaCollectionTable
+          datalist={<DataList id="fcriteria" options={this.state.dataList} />}
+          columnHeaders={this.columnHeaders}
+          allTeas={this.props.teas.allTeas}
+          teaIDs={this.state.sortedIDs}
+          formControls={this.state.formControls}
+          filtered={this.state.filtered}
+          handleDeleteClick={this.handleDeleteClick}
+          handleSortClick={this.handleSortClick}
+          handleFilterClick={this.handleFilterClick}
+          handleClearFilterClick={this.handleClearFilterClick}
+          filterDropdownChangeHandler={this.filterDropdownChangeHandler}
+          filterInputChangeHandler={this.filterInputChangeHandler}
+          sortColumnHandler={this.sortColumnHandler}
+        />
+      )
+    );
+  }
+}
+
+const mapStateToProps = state => {
+  return {
+    teas: state.teas,
+    teaTypes: state.teaTypes,
+    userID: state.auth.user.id
+  };
+};
+
+const mapDispatchToProps = dispatch => ({
+  handleDelete: teaID => {
+    dispatch(deleteTea(teaID));
+  },
+  getTeaList: userID => {
+    dispatch(getTeas(userID));
+  }
+});
```

Comment:
This can be simplified to `const mapDispatchToProps = { handleDelete: deleteTea, getTeaList: getTeas };`

It could be simplified even further if you didn't rename the actions.